tiledb/sm/filesystem/vfs.cc

```diff
@@ -144,6 +145,45 @@ Status VFS::create_dir(const URI& uri) const {
   STATS_FUNC_OUT(vfs_create_dir);
 }
 
+Status VFS::dir_size(const URI& dir_name, uint64_t* dir_size) const {
+  // Sanity check
+  bool is_dir;
+  RETURN_NOT_OK(this->is_dir(dir_name, &is_dir));
+  if (!is_dir)
+    return LOG_STATUS(Status::VFSError(
+        std::string("Cannot get directory size; Input '") +
+        dir_name.to_string() + "' is not a directory"));
+
+  // Get all files in the tree rooted at `dir_name`
+  std::vector<URI> files;
```

Comment:
why create an entire list of files (recursively) and then get the sizes as opposed to just cumulative summing the sizes lazily?